OpenNefia.Content/CharaMake/CharaMakeAttributeRerollLayer.cs

```diff
@@ -0,0 +1,239 @@
+
+                switch (Data.Data)
+                {
+                    default:
+                    case AttributeRerollChoice choice:
+                        Icon = new AttributeIcon("");
+                        break;
+                    case PrototypeId<SkillPrototype> id:
+                        Data.AmountText = AmountText;
+                        Icon = new AttributeIcon(id.ToString());
+                        Data.Amount = new System.Random().Next(1, 14);
+                        break;
+                }
+            }
+
+            public override void SetSize(int width, int height)
+            {
+                height = 22;
+                base.SetSize(width, height);
+            }
+
+            public override void SetPosition(int x, int y)
+            {
+                base.SetPosition(x, y);
+                Icon.SetPosition(x + 150, y + 9);
+                AmountText.SetPosition(x + 162, y);
+                LockedText.SetPosition(x + 185, y + 3);
+            }
+
+            public override void Draw()
+            {
+                base.Draw();
+                if (Data.Locked)
+                    LockedText.Draw();
+                Icon.Draw();
+                AmountText.Draw();
+            }
+        }
+
+        [Localize] private UiWindow Window = new();
+        [Localize] private UiTextTopic AttrTopic = new();
+        [Localize] private UiWrapText AttrInfo;
+        private UiText LockAmt = new();
+        private UiList<AttributeRerollData> List;
+
+        private int LockCount = 2;
+
+        public CharaMakeAttributeRerollLayer()
+        {
+            AttrInfo = new UiWrapText(115, UiFonts.LockInfo);
+            List = new UiList<AttributeRerollData>();
+            LockAmt = new UiText(UiFonts.LockInfo);
+            AttrInfo.Text = Loc.GetString("Elona.CharaMake.AttributeReroll.AttrInfo");
+            SetLockCountText();
+            AddChild(List);
+        }
+
+        public override void Initialize(CharaMakeData args)
+        {
+            base.Initialize(args);
+            var data = AttributeIds.Select(x => new AttributeRerollCell(new PrototypeId<SkillPrototype>(x))).ToList();
+            data.InsertRange(0, new[]
+            {
+                new AttributeRerollCell(AttributeRerollChoice.Reroll),
+                new AttributeRerollCell(AttributeRerollChoice.Proceed)
+            });
+            List.AddRange(data);
+            List.EventOnActivate += (_, args) =>
+            {
+                switch(args.SelectedCell.Data.Data)
+                {
+                    case AttributeRerollChoice choice:
+                        switch (choice)
+                        {
+                            case AttributeRerollChoice.Proceed:
+                                Finish(new CharaMakeResult(new Dictionary<string, object>
+                                {
+                                    { "attributes", List.Select(x => x.Data)
```

Comment:
Nitpick: Maybe move these keys into `public const string`s per layer class?

Reply:
done in 76f6762